client/landing/stepper/declarative-flow/link-in-bio-tld.ts

```diff
@@ -0,0 +1,149 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { translate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { recordFullStoryEvent } from 'calypso/lib/analytics/fullstory';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import wpcom from 'calypso/lib/wp';
+import {
+	clearSignupDestinationCookie,
+	setSignupCompleteSlug,
+	persistSignupDestination,
+	setSignupCompleteFlowName,
+} from 'calypso/signup/storageUtils';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { USER_STORE, ONBOARD_STORE } from '../stores';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import DomainsStep from './internals/steps-repository/domains';
+import LaunchPad from './internals/steps-repository/launchpad';
+import LinkInBioSetup from './internals/steps-repository/link-in-bio-setup';
+import PatternsStep from './internals/steps-repository/patterns';
+import PlansStep from './internals/steps-repository/plans';
+import Processing from './internals/steps-repository/processing-step';
+import SiteCreationStep from './internals/steps-repository/site-creation-step';
+import type { Flow, ProvidedDependencies } from './internals/types';
+
+const linkInBio: Flow = {
+	name: LINK_IN_BIO_TLD_FLOW,
+	get title() {
+		return translate( 'Link in Bio' );
+	},
+	useSteps() {
+		useEffect( () => {
+			recordTracksEvent( 'calypso_signup_start', { flow: this.name } );
+			recordFullStoryEvent( 'calypso_signup_start_link_in_bio', { flow: this.name } );
+		}, [] );
+
+		return [
+			{ slug: 'domains', component: DomainsStep },
+			{ slug: 'patterns', component: PatternsStep },
+			{ slug: 'linkInBioSetup', component: LinkInBioSetup },
+			{ slug: 'plans', component: PlansStep },
+			{ slug: 'siteCreationStep', component: SiteCreationStep },
+			{ slug: 'processing', component: Processing },
+			{ slug: 'launchpad', component: LaunchPad },
+		];
+	},
+
+	useStepNavigation( _currentStepSlug, navigate ) {
+		const flowName = this.name;
+		const { setStepProgress } = useDispatch( ONBOARD_STORE );
+		const flowProgress = useFlowProgress( { stepName: _currentStepSlug, flowName } );
+		const siteSlug = useSiteSlug();
+		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
+		const locale = useLocale();
+
+		setStepProgress( flowProgress );
+
+		// trigger guides on step movement, we don't care about failures or response
+		wpcom.req.post(
+			'guides/trigger',
+			{
+				apiNamespace: 'wpcom/v2/',
+			},
+			{
+				flow: flowName,
+				step: _currentStepSlug,
+			}
+		);
+
+		const logInUrl =
+			locale && locale !== 'en'
+				? `/start/account/user/${ locale }?variationName=${ flowName }&pageTitle=Link%20in%20Bio&redirect_to=/setup/${ flowName }/patterns`
+				: `/start/account/user?variationName=${ flowName }&pageTitle=Link%20in%20Bio&redirect_to=/setup/${ flowName }/patterns`;
```

Comment:
Do we need to do this manually?

Reply:
I suppose we don't. This PR focuses on making the flow work first, so I simply refer to how `link-in-bio.ts` does it for now. It can be improved as a separate PR for sure :)

Reply:
Yes @alshakero, this is how we are doing in the LiB flow as well, I think it can be good here for this PR :)